### PR TITLE
Snapshot test cleanup

### PR DESCRIPTION
**Pull request by eczarny**
To help reduce the amount of friction currently needed to record snapshot tests let's remove the reference images that can be removed. This pull request includes the following changes:

1. Limit iOS builds to a 64-bit iOS 9 device (the iPhone 5s simulator)
2. Remove all iOS 8 reference images
3. Remove all 32-bit iOS 9 reference images

Internally the 64-bit iOS 9 reference images are still needed; however, once internal builds have migrated to Xcode 8 and iOS 10 we can remove the remaining iOS 9 reference images. Moving forward we should only include reference images for the latest iOS release.

To reduce the time spent in each build the following additional changes have been made:

1. Specify the platform when running `carthage bootstrap` (e.g. tvOS for the `ci-componentkit-tvos` build)
2. Skip `carthage bootstrap` for the `ci-wildeguess-ios` build

These changes mean less time spent bootstrapping unnecessary test dependencies.

### CODE REVIEW

**Comment by facebook-github-bot-1** [comment]
By analyzing the blame information on this pull request, we identified @conradev to be a potential reviewer.

**Comment by facebook-github-bot-1** [comment]
@eczarny updated the pull request - [view changes](https://github.com/facebook/componentkit/pull/647/files/5d66aa7e39b708a2a2e3c302e6da55e00c11c212..1e713592c2fb546e003e2ebd2774e81c42417d36)

**Comment by facebook-github-bot-8** [comment]
@eczarny updated the pull request - [view changes](https://github.com/facebook/componentkit/pull/647/files/1e713592c2fb546e003e2ebd2774e81c42417d36..15685c0104d7422c7dbe017a63401cc76fb66dc8)

**Comment by facebook-github-bot-1** [comment]
@eczarny updated the pull request - [view changes](https://github.com/facebook/componentkit/pull/647/files/15685c0104d7422c7dbe017a63401cc76fb66dc8..92761aa14259e7e597683c8d67a42241f1fe0dd8)

**Comment by facebook-github-bot-9** [comment]
@eczarny updated the pull request - [view changes](https://github.com/facebook/componentkit/pull/647/files/92761aa14259e7e597683c8d67a42241f1fe0dd8..01206ecf4afeaa7bd321f0fd54472b053b747d80)

**Comment by eczarny** [comment]
Currently fighting with Travis CI. Testing changes to the build isn't easy.

**Comment by facebook-github-bot-5** [comment]
@eczarny updated the pull request - [view changes](https://github.com/facebook/componentkit/pull/647/files/01206ecf4afeaa7bd321f0fd54472b053b747d80..52fd21c57baf1ceb1ea5c0dac994add0e8b1a9bb)

**Comment by eczarny** [comment]
Ready for review.

File: .travis.yml [hunk 1]
```diff
@@ -4,17 +4,15 @@ language: objective-c
 
 install:
   - gem install xcpretty -N --no-ri --no-rdoc
-  - carthage bootstrap
```
Comment (eczarny) [inline .travis.yml]: No longer needed here since `carhage bootstrap` is handled by the build script.

File: .travis.yml [hunk 2]
```diff
@@ -4,17 +4,15 @@ language: objective-c
...
 
 script:
-  - set -o pipefail
-  - xcodebuild $XCODEBUILD_CONFIGURATION | xcpretty
+  - ./build.sh $BUILD_SUBCOMMAND
```
Comment (eczarny) [inline .travis.yml]: The build script provides a bit more flexibility.

File: build.sh [hunk 3]
```diff
@@ -29,29 +31,30 @@ function ci() {
 }
 
 function ios_ci() {
-  ci $1 $1 iphonesimulator9.3 "platform=iOS Simulator,OS=9.3,name=iPhone 5" $2
+  ci $1 $2 iphonesimulator9.3 "platform=iOS Simulator,OS=9.3,name=iPhone 5s" $3
```
Comment (eczarny) [inline build.sh]: The iPhone 5s is a 64-bit device that is compatible for the current set of reference images.

File: build.sh [hunk 4]
```diff
@@ -29,29 +31,30 @@ function ci() {
...
 }
 
 function tvos_ci() {
-  ci $1 $1AppleTV appletvsimulator "platform=tvOS Simulator,OS=9.2,name=Apple TV 1080p" $2
+  ci $1 $2 appletvsimulator9.2 "platform=tvOS Simulator,OS=9.2,name=Apple TV 1080p" $3
```
Comment (eczarny) [inline build.sh]: Let's be explicit with the SDK.

File: build.sh [hunk 5]
```diff
@@ -29,29 +31,30 @@ function ci() {
...
-  pushd Examples/WildeGuess
-  ios_ci WildeGuess build
-  popd
+if [ "$MODE" = "ci-componentkit-tvos" ]; then
+  carthage bootstrap --platform tvOS
+  tvos_ci ComponentKit.xcodeproj ComponentKitAppleTV test
+fi
 
+if [ "$MODE" = "ci-wildeguess-ios" ]; then
+  ios_ci Examples/WildeGuess/WildeGuess.xcodeproj WildeGuess build
```
Comment (eczarny) [inline build.sh]: WildeGuess doesn't include any tests so `carthage bootstrap` isn't necessary.

**Comment by ocrickard** [comment]
OMG thank you. Let's merge this ASAP.

**Comment by eczarny** [comment]
Once CI passes I'll get this merged.

**Comment by facebook-github-bot-6** [comment]
@eczarny updated the pull request - [view changes](https://github.com/facebook/componentkit/pull/647/files/52fd21c57baf1ceb1ea5c0dac994add0e8b1a9bb..5805f562bbc26e71b5579ab845c93fe73b51b2fa)

**Pull request merged**